docs/src/rules/no-await-in-loop.md

```diff
@@ -37,6 +37,37 @@ async function foo(things) {
 }
 ```
 
+Also, if a loop iterates over `Promise` rejections with `await`, then after the first promise is rejected, it immediately throws an error, stopping the loop's execution before handling the rest of the promises. This leads to unhandled rejections for any promises that haven’t been awaited yet.
+
+```js
+async function bar(things) {
+  const arrayOfRejections = new Array(things).fill(undefined).map((_, index) => Promise.reject(new Error("rejected " + index)));
+  const result = [];
+  for (const promise of arrayOfRejections) {
+    // Bad: after the rejection of first promise, loop's execution stops, leaving the rest of the promise rejections floating and unhandled.
+    const promisedValue = await promise;
+    result.push(promisedValue);
+  }
+
+  return result;
+}
+```
+
+This function should be refactored as shown:
+
+```js
+async function bar(things) {
+  const arrayOfRejections = new Array(things).fill(undefined).map((_, index) => Promise.reject(new Error("rejected " + index)));
+  const result = [];
+  // Good: all the promises are immediately rejected so there are no floating unhandled rejections.
```

Comment:
```suggestion
  // Good: if any of the promises fail, all others are rejected immediately, so there are no floating unhandled rejections.

```

Reply:
> // Good: if any of the promises fail, all others are rejected immediately, so there are no floating unhandled rejections.

This suggestion isn't quite right. The other promises' resolutions aren't changed in any way. They just don't cause unhandled rejections https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

```suggestion
  // Good: if any of the promises rejects, the Promise.all() call will reject, and any rejections in the other promises will not be treated as unhandled rejections
```

Reply:
Updated!